### ISSUE TITLE
Seriously Aggressive Bots

### ISSUE THREAD

**Issue by Matuchkin** [RF-Stuff/Bugs-Suggestions#140]
Currently, bots travel to a single point, at which they position themselves and intermittently fire at other bots. After that is done, they sit there and defend the point for most of the game. This suggestion will outline my train of thought of how these bots can be turned from what they are now to more lethal and "human" soldiers:

a) Panic.
          - When under heavy fire, under low cover, and in small numbers, bots will enter a panic mode.
          - Volume of fire, proximity of friendly bots, and cross-sectional area of cover will be taken into 
            calculation.
          Depending on the situation, a bot may cower, rout (run away), or fire frantically (very rapidly and 
          inaccurately).

b) Reaction.
          - The bots will be extremely attentive and reactive; so much as a burst of fire in their direction, and 
            they (along with any friendly bot in their vicinity) will suddenly dive for cover, _instantly_ 
            beginning a rapid, defensive course of action.
          - They will act either defensively or offensively.
DEFENSIVE: Bots sprint from cover to cover, around their zone. Fire is directed with precision, towards advancing enemies, firing en-masse rather than at single individuals (i.e. slow, sweeping movements in the direction of the enemies. Any explosives are used for area damage rather than to kill, and are fired at the largest / more dense gatherings of robots. Armored vehicles are at a standstill, firing away at every opportunity. Any aircraft in the vicinity instantly start CAS activity over the enemy.

OFFENSIVE: Bots are dedicated to moving to the capture point. Sprint from cover to cover, firing in single or double bursts at the closest enemy bot to them. Anyone with an explosive or area damage weapon is prioritized, as are armored vehicles or aircraft. Any friendly armored vehicles are moving forward, with the intent of i) firing at any enemy that moves, ii) rolling in front of the infantry to protect them. Any aircraft in the vicinity start bombarding the enemy's defenses.

c) ... perhaps some strategy?
          - For example: when finding a force larger than theirs, the enemy will either assume a defensive 
            position or retreat while firing back, to keep the latter at long range.
          - AI aircraft will be called to positions in which the most fighting is going on, or to the largest 
            gathering of currently-firing enemies on the map.

**Comment by Matuchkin** [comment]
And, for the record, I would like to add a crucial aspect: in current Ravenfield, bots generally suck at attacking positions, only advancing at a trickle. Thus, battles usually stagnate and never actually advance. This will change: when any position becomes an objective, the AI system will throw all it can against that position, and it will be very aggressive. Obviously, this will utilise methods used in the above post.

**Comment by Axeman76** [comment]
I agree with these suggestions, I think they would make the game even more enjoyable.

**Comment by Jack-Attack227** [comment]
Adding onto Seriously Aggressive Bots:
![7ers2c2](https://user-images.githubusercontent.com/38716200/41805710-a1197482-76f2-11e8-89a7-ae0ffca306d4.jpg)
![images3p2ilv3o](https://user-images.githubusercontent.com/38716200/41805716-b3d4219e-76f2-11e8-9c7c-076b0ce92dc3.jpg)



Suppression: 
It occurs when bullets land or are within a few metres of you or the bots. 
When it happens to bots they would enter the panic mode. The effect would be most effective with light or heavy machine guns or sniper rifles. For you when it happened a blur effect would happen and the higher calibre and closer the bullets the intenser the blur would get. If you or the AI are getting supressed you become slower and less accurate. Also AI that were using LMG/HMGs or sniper rifles will use this to their advantage they will stay back and fire at where the AI were last seen.   

Bipods: 
Will be implement this will help AI supress enemies.  

Prone:
For most of this to work prove and diving into cover will need to be implemented. 

Explosives: 
Loud ringing and blur effect would happen when and explosion goes off near you, AI if they see a grenade near them will either run away or dive onto the grenade this will only kill the bot that dived onto it and save the other bots.    

Vehicles:
Ai will run away and find cover if they see a vehicle and they have to anti-tank weapons and if planes or helicopters are coming towards AI they will scatter and find cover.   

Positions:
Ai will hold key positions even if they aren’t actual points, for instance if there is a hill that overlooks a point AI will hold the hill, these point could be placed by map makers.

**Comment by Matuchkin** [comment]
Wow... uh... I didn't expect such an in depth reply. I think what you said
in terms of suppression is a start, and I like the idea with bipods. I also
think that units with no explosive weaponry should stay away from tanks,
helis, etc. The position holding is nice, though I am not sure how we can
implement that in Unity.
